Lib/test/test_code_module.py

```diff
@@ -39,19 +39,47 @@ def setUp(self):
         self.mock_sys()
 
     def test_ps1(self):
-        self.infunc.side_effect = EOFError('Finished')
+        self.infunc.side_effect = [
+            "import code",
+            "code.sys.ps1",
+            EOFError('Finished')
+        ]
         self.console.interact()
-        self.assertEqual(self.sysmod.ps1, '>>> ')
+        output = ''.join(''.join(call[1]) for call in self.stdout.method_calls)
+        self.assertIn('>>> ', output)
+        self.assertFalse(hasattr(self.sysmod, 'ps1'))
```

Comment:
`assertNotHasAttr`

Reply:
I think it depends on whether we consider this a bug fix and backport it. `assertNotHasAttr` is new in 3.14 which would cause issues when we backport the test. If we only want to fix this in 3.14 (which is fine by me because I only need this in 3.14), then sure we can use the latest method.

@gvanrossum do you think we should backport this? Or just leave it be for the previous versions.